compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -994,6 +995,40 @@ static FailureOr<Type> nonWideningLinalgElementType(linalg::LinalgOp op) {
   return inputAndOutputElementTypes[0];
 }
 
+/// Utility to compute the tile sizes for AArch64 Neon/SVE.
+static void getMatmulAArch64NeonSVEVectorSizes(
+    func::FuncOp entryPointFn, linalg::LinalgOp op, int64_t vectorSize,
+    SmallVectorImpl<int64_t> &sizes, SmallVectorImpl<bool> &scalableSizeFlags) {
+  getDefaultMatmulVectorSizes(op, vectorSize, sizes, scalableSizeFlags);
+
+  // Find the smallest type size in the matmul.
+  SmallVector<Type> matmulTypes;
+  auto operandTypes = op->getOperandTypes();
+  matmulTypes.append(operandTypes.begin(), operandTypes.end());
+  auto resultTypes = op->getResultTypes();
+  matmulTypes.append(resultTypes.begin(), resultTypes.end());
+
+  int64_t minSize = std::numeric_limits<int64_t>::max();
+  for (Type mmType : matmulTypes) {
+    if (auto shType = dyn_cast<ShapedType>(mmType))
+      mmType = shType.getElementType();
+
+    if (mmType.isSignlessIntOrFloat())
+      minSize = std::min<int64_t>(minSize, mmType.getIntOrFloatBitWidth());
+  }
+
+  LLVM_DEBUG(KD_DBGS() << "Smallest type found: " << minSize << " bits\n");
+
+  if (minSize < 0 || minSize == std::numeric_limits<int64_t>::max())
+    return;
+
```

Comment:
I think we can remove the check because it is invalid if a LinalgOp does not have any operands. And the bitwidth is guaranteed to be greater than 0?

Reply:
Let me turn it into an assert. I'm worried that some present or future corner cases can make it through (i.e., non int/float cases, vector element types, some of the 0-D or scalar case, etc.). Just to fall on the safe side.

Reply:
SGTM!